crates/storage/storage-api/src/state.rs

```diff
@@ -91,6 +91,13 @@ pub trait TryIntoHistoricalStateProvider {
     ) -> ProviderResult<StateProviderBox>;
 }
 
+/// Trait implemented for database providers that can be converted into a latest state provider
+/// reference.
+pub trait ToLatestStateProviderRef {
```

Comment:
I think generally we would use `Into` when the tx is being consumed. I chose `To` because I believe it best fits the guidelines [here](https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv). However, I'm not strongly opinionated and happy to change this to `AsLatestStateProviderRef`.

Reply:
Changed to `AsLatestStateProviderRef` as suggested.